src/pages/ThorChainLP/queries/hooks/useSendThorTx.tsx

```diff
@@ -0,0 +1,458 @@
+type ThorfiAction = `${ThorfiActionBase}Enter` | `${ThorfiActionBase}Exit` | 'ThorchainSwap'
+
+type Props = {
+  amountCryptoBaseUnit: string | undefined
+  dustAmountCryptoBaseUnit: string | undefined
+  assetId: AssetId
+  // TODO(gomes): we should build this here too, but until this is properly implemented for all domains we're just passing it
+  // to keep things simple
+  memo: string | undefined
+  accountId: AccountId
+  transactionType: 'MsgDeposit' | 'EvmCustomTx' | 'Send' | undefined
+  assetAddress: string | null
+  thorfiAction: ThorfiAction
+  // Indicates whether the consumer is currently submitting, meaning we should stop refetching
+  isSubmitting?: boolean
+}
+
+export const useSendThorTx = ({
+  amountCryptoBaseUnit,
+  dustAmountCryptoBaseUnit,
+  assetId, // i.e the AssetId of the Tx, either native asset, or token to be sent from the contract, assuming allowance has been set
+  memo,
+  accountId,
+  transactionType,
+  assetAddress,
+  thorfiAction,
+  isSubmitting = false,
+}: Props) => {
+  // TODO(gomes): savers sometimes also use dust amounts, ensure this works for them
+  const shouldUseDustAmount = thorfiAction === 'ThorchainLpExit'
+  const [txId, setTxId] = useState<string | null>(null)
+  const [serializedTxIndex, setSerializedTxIndex] = useState<string | null>(null)
+  const wallet = useWallet().state.wallet
+  const toast = useToast()
+  const translate = useTranslate()
+
+  const asset = useAppSelector(state => selectAssetById(state, assetId))
+  const feeAsset = useAppSelector(state =>
+    selectFeeAssetByChainId(state, fromAssetId(assetId).chainId),
+  )
+
+  const accountNumberFilter = useMemo(() => ({ assetId, accountId }), [accountId, assetId])
+  const accountNumber = useAppSelector(s => selectAccountNumberByAccountId(s, accountNumberFilter))
+  const selectedCurrency = useAppSelector(selectSelectedCurrency)
+
+  const amountCryptoPrecision = useMemo(
+    () => fromBaseUnit(amountCryptoBaseUnit ?? 0, asset?.precision ?? 0),
+    [amountCryptoBaseUnit, asset],
+  )
+  const dustAmountCryptoPrecision = useMemo(
+    () => fromBaseUnit(dustAmountCryptoBaseUnit ?? 0, asset?.precision ?? 0),
+    [dustAmountCryptoBaseUnit, asset],
+  )
+
+  const { data: inboundAddressData } = useQuery({
+    ...reactQueries.thornode.inboundAddresses(),
+    staleTime: 60_000,
+    select: data => selectInboundAddressData(data, assetId),
+    enabled: !!assetId,
+  })
+
+  const estimateFeesArgs = useMemo(() => {
+    if (!assetId || !feeAsset || !wallet || !asset) return undefined
+
+    const amountCryptoBaseUnit = toBaseUnit(amountCryptoPrecision, asset.precision)
+
+    // TODO(gomes): when consuming this in THORCHain savers, ensure we do not send dust amount instead of depositWithExpiry() Txs for EVM chains
+    const dustAmountCryptoBaseUnit =
+      THORCHAIN_SAVERS_DUST_THRESHOLDS_CRYPTO_BASE_UNIT[feeAsset.assetId] ?? '0'
+    const dustAmountCryptoPrecision = fromBaseUnit(dustAmountCryptoBaseUnit, feeAsset.precision)
```

Comment:
this would essentially replace L:96 for now and used throughout the hook as needed

Reply:
24dd66ff makes things tidier